pandas/_libs/missing.pyx

```diff
@@ -434,6 +452,32 @@ class NAType(C_NAType):
 
     __rxor__ = __xor__
 
+    __array_priority__ = 1000
+    _HANDLED_TYPES = (np.ndarray, numbers.Number, str, np.bool_)
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        types = self._HANDLED_TYPES + (NAType,)
+        for x in inputs:
+            if not isinstance(x, types):
+                return NotImplemented
+
+        if method != "__call__":
+            raise ValueError(f"ufunc method '{method}' not supported for NA")
+        result = maybe_dispatch_ufunc_to_dunder_op(
+            self, ufunc, method, *inputs, **kwargs
+        )
+        if result is NotImplemented:
+            # TODO: this is wrong for binary, ternary ufuncs. Should handle shape stuff.
```

Comment:
When does it happen that the dispatch function returns NotImplemented?

Reply:
The comment just needs to be removed. It’s handled correctly I think. 

It’s not implemented for ufuncs that aren’t dunder ops.